js/src/annotations/osd-region-draw-tool.js

```diff
@@ -0,0 +1,424 @@
+
+    deleteShape: function() {
+      var _this = this;
+      if (_this.svgOverlay.hoveredPath) {
+        var oaAnno = null;
+        for (var key in _this.annotationsToShapesMap) {
+          if (_this.annotationsToShapesMap.hasOwnProperty(key)) {
+            var shapeArray = _this.annotationsToShapesMap[key];
+            for (var idx = 0; idx < shapeArray.length; idx++) {
+              if (shapeArray[idx].name == _this.svgOverlay.hoveredPath.name) {
+                oaAnno = shapeArray[idx].data.annotation;
+                if (shapeArray.length == 1) {
+                  if (!window.confirm("Do you want to delete this shape and annotation?")) {
+                    return false;
+                  }
+                  jQuery.publish('annotationDeleted.' + _this.parent.windowId, [oaAnno['@id']]);
+                  this.svgOverlay.removeFocus();
+                  return true;
+                } else {
+                  if (!window.confirm("Do you want to delete this shape?")) {
+                    return false;
+                  }
+                  shapeArray.splice(idx, 1);
+                  oaAnno.on.selector.value = _this.svgOverlay.getSVGString(shapeArray);
+                  jQuery.publish('annotationUpdated.' + _this.parent.windowId, [oaAnno]);
+                  this.svgOverlay.removeFocus();
+                  return true;
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+
+    saveEditedShape: function() {
+      var _this = this;
+      var oaAnnos = [];
+      for (var key in _this.annotationsToShapesMap) {
+        if (_this.annotationsToShapesMap.hasOwnProperty(key)) {
+          var shapeArray = _this.annotationsToShapesMap[key];
+          var annotationShapesAreEdited = false;
+          for (var i = 0; i < _this.svgOverlay.editedPaths.length; i++) {
+            for (var idx = 0; idx < shapeArray.length; idx++) {
+              if (shapeArray[idx].name == _this.svgOverlay.editedPaths[i].name) {
+                oaAnnos.push(shapeArray[idx].data.annotation);
+                oaAnnos[oaAnnos.length - 1].on.selector.value = _this.svgOverlay.getSVGString(shapeArray);
+                annotationShapesAreEdited = true;
+                break;
+              }
+            }
+            if (annotationShapesAreEdited) {
+              break;
+            }
+          }
+        }
+      }
+      for (var j = 0; j < oaAnnos.length; j++) {
+        jQuery.publish('annotationUpdated.' + _this.parent.windowId, [oaAnnos[j]]);
+      }
+      this.svgOverlay.restoreEditedShapes();
+    },
+
+    render: function() {
+      this.svgOverlay.paperScope.project.clear();
+      var _this = this;
+      _this.annotationsToShapesMap = {};
+      var deferreds = jQuery.map(this.list, function(annotation) {
+        var deferred = jQuery.Deferred();
+        if (typeof annotation.on === 'object') {
+          var shapeArray = _this.svgOverlay.parseSVG(annotation.on.selector.value, annotation);
```

Comment:
@stanislavkostadinov For backwards compatibility (and to handle simple rectangular, non SVG annotations) this needs to be able to handle annotations where ```annotation.on.selector.value``` can look like this: ```xywh=2164,1161,695,368```.  This is the ```media-frag``` Fragment Selector: http://www.openannotation.org/spec/core/specific.html#FragmentSelector.